packages/kbn-dependency-usage/src/cli.ts

```diff
@@ -0,0 +1,340 @@
+    : {
+        name: 'external-deps',
+        severity: 'info',
+        from: {
+          path: paths.map((path) => `^${path}`),
+        },
+        to: {
+          path: '^node_modules',
+        },
+      };
+  const { cruise } = await import('dependency-cruiser');
+  const result = await cruise(paths, {
+    ruleSet: {
+      // @ts-ignore
+      forbidden: [captureRule],
+    },
+    doNotFollow: {
+      path: 'node_modules',
+    },
+    extensions: ['.ts', '.tsx'],
+    focus: '^node_modules',
+    exclude: {
+      path: [
+        '(^|/)target($|/)',
+        '^kbn',
+        '^@kbn',
+        '@elastic',
+        '^.buildkite',
+        '^test',
+        '^docs',
+        '^dev_docs',
+        '^examples',
+        '^scripts',
+        '^bazel',
+        '^x-pack/test',
+        '^x-pack/test_serverless'
+      ],
+    },
+    onlyReachable: paths.map((path) => `^${path}`).join('|'),
+    includeOnly: ['^node_modules', ...paths.map((path) => `^${path}`)],
+    validate: true,
+    collapse: collapseRules.join('|'),
+  });
+
+  return result;
+};
+
+async function identifyDependencyUsage(
+  dependencyName: string,
+  paths: string[],
+  {
+    groupBy,
+    summary,
+    collapseDepth,
+  }: { groupBy?: string; summary?: boolean; collapseDepth: number }
+) {
+  const result = await invokeCruise({ dependencyName, paths, summary, collapseDepth });
+
+  if (typeof result.output === 'string') {
+    throw new Error('Unexpected string output from cruise result');
+  }
+  const { violations } = result.output.summary;
+
+  if (dependencyName) {
+    const dependencyRegex = new RegExp(`node_modules/${dependencyName}`);
+
+    const dependentsList = result.output.modules.reduce<Record<string, string[]>>(
+      (acc: { [x: string]: string[] }, { source, dependents }: any) => {
+        if (!dependencyRegex.test(source)) {
+          return acc;
+        }
+
+        const transformedDependencyName = source.split('/')[1];
+        if (!acc[transformedDependencyName]) {
+          acc[transformedDependencyName] = [];
+        }
+
+        acc[transformedDependencyName].push(...dependents);
+
+        return acc;
+      },
+      {} as Record<string, string[]>
+    );
+
+    return {
+      modules: [...new Set(violations.map(({ from }) => from))],
+      ...(!summary && { dependents: dependentsList }),
+    };
+  }
+
+  if (groupBy === 'owner') {
+    const byOwners = groupFilesByOwners(violations);
+
+    return byOwners;
+  } else {
+    return groupByPackage(violations);
+  }
+}
+
+// Added for comparison, grouping by owners/package is not fully supported when using madge
+const identifyDependencyUsageWithMadge = async (dependencyName: string, paths: string[]) => {
+  const response = await madge(paths, {
+    fileExtensions: ['ts', 'tsx', 'js', 'jsx'],
+    includeNpm: true,
+    excludeRegExp: [
+      '^node_modules',
+      '(^|/)target($|/)',
+      '^kbn',
+      '^@kbn',
+      '@elastic',
+      '^.buildkite',
+      '^test',
```

Comment:
> 1/ Is this supposed to exclude the test directory?

This is probably because madge treats everything relative, so when `x-pack/test` becomes a base dir it can exclude `test` inside `x-pack/test`, so this one `x-pack/test/test`, but not the root folder itself

> 2/ If so, is there a reason we want to exclude test directories from the scan?
We can add them, excluded them for testing purpose since run for whole kibana takes time

Reply:
Gotcha, thanks. I think it makes sense to allow these test directories to be scanned